### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Description
This PR adds several improvements by:
  - Removing the license information from the PR template file (it's annoying to have this in github when opening a PR)
  - Adjusting the several config files to how they are in other projects
  - Adding Jetbrains Copyright configuration
  - Adding prettier (and applying it)
  - Pin dev dependencies
  - Lint test files correctly

### Steps

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/markdown-it-image-size/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .github/workflows/lint.yml:1]:
```suggestion
# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .github/workflows/reuse.yml [hunk 2]
```diff
@@ -1,15 +1,19 @@
-# SPDX-FileCopyrightText: 2020 The HedgeDoc developers (see AUTHORS file)
+# SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .github/workflows/reuse.yml:1]:
```suggestion
# SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .prettierignore.license [hunk 3]
```diff
@@ -0,0 +1,4 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierignore.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .prettierrc.json.license [hunk 4]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .prettierrc.json.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .reuse/dep5 [hunk 5]
```diff
@@ -2,3 +2,15 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
 Upstream-Name: markdown-it-image-size
 Upstream-Contact: The HedgeDoc developers <license@hedgedoc.org>
 Source: https://github.com/hedgedoc/markdown-it-image-size
+
+Files: .github/pull_request_template.md
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:7]:
```suggestion
Copyright: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: .reuse/dep5 [hunk 6]
```diff
@@ -2,3 +2,15 @@ Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
...
+License: CC-BY-SA-4.0
+
+Files: .yarn/**
+Copyright: Yarn contributors
+License: LicenseRef-yarn
+
+Files: .idea/**
+Copyright: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline .reuse/dep5:15]:
```suggestion
Copyright: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: package.json [hunk 7]
```diff
@@ -13,9 +13,10 @@
   ],
   "scripts": {
     "test": "jest",
-    "build": "tsc",
-    "prepublish": "tsc",
-    "lint": "eslint src --ext .ts"
+    "build": "tsc -p tsconfig.build.json",
+    "prepublish": "rm -rf dist && yarn lint && yarn build && yarn test",
+    "lint": "eslint --ext .ts src",
+    "lint:fix": "eslint --fix --ext .ts src"
```
Comment (DerMolly) [inline package.json:19]: Should we add `format` here?

File: tsconfig.build.json.license [hunk 8]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline tsconfig.build.json.license:1]:
```suggestion
SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
```

File: yarn.lock.license [hunk 9]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
```
Comment (DerMolly) [inline yarn.lock.license:1]: 2022

File: package.json (hunk 7 above)
Comment (mrdrogdrog) [reply package.json:19]: I added the eslint plugin for prettier. Therefore eslint checks prettier and fix reformats the code.

**Review by DerMolly — APPROVED** [review]

**Pull request merged**